eos_api.hpp

```diff
@@ -1121,35 +1121,115 @@ const uint8_t proofStorage_IPFS = 0x01;
+ *                   Definition                   *
  **************************************************/
-std::string vector_to_string(std::vector<uint8_t> v)
+struct [[eosio::table, eosio::contract("oraclizeconn")]] snonce
+{
+    name sender;
+    uint32_t nonce;
+
+    uint64_t primary_key() const { return sender.value; }
+};
+
+struct [[eosio::table, eosio::contract("oraclizeconn")]] cbaddr
+{
+    name sender;
+
+    uint64_t  primary_key() const { return sender.value; }
+};
+
+struct [[eosio::table]] scommitment
+{
+    name shortqueryid;
+    capi_checksum256 queryid;
+    capi_checksum256 commitment;
+
+    uint64_t primary_key() const { return shortqueryid.value; }
+};
+
+struct [[eosio::table, eosio::contract("oraclizeconn")]] spubkey
+{
+    name key;
+    capi_checksum256 randomDS_lastSessionPubkeyHash;
+
+    uint64_t primary_key() const { return key.value; }
+    capi_checksum256 get_randomDS_lastSessionPubkeyHash() const { return randomDS_lastSessionPubkeyHash; }
+};
+
+struct [[eosio::table]] queryid
+{
+    name key;
+    capi_checksum256 qid;
+    uint8_t active;
+
+    uint64_t primary_key() const { return key.value; }
+};
+
+typedef eosio::multi_index<name("snonce"), snonce> ds_snonce;
+typedef eosio::multi_index<name("cbaddr"), cbaddr> ds_cbaddr;
+typedef eosio::multi_index<name("scommitment"), scommitment> ds_scommitment;
+typedef eosio::multi_index<name("spubkey"), spubkey> ds_spubkey;
+typedef eosio::multi_index<name("queryid"), queryid> ds_queryid;
+
+
+/**************************************************
+ *                      MACRO                     *
+ *                   Definitions                  *
+ **************************************************/
+#ifndef ORACLIZE_PAYER
+#define ORACLIZE_PAYER _self
+#endif // ORACLIZE_PAYER
+
+#define oraclize_query(...) __oraclize_query(ORACLIZE_PAYER, __VA_ARGS__, _self)
+#define oraclize_newRandomDSQuery(...) __oraclize_newRandomDSQuery(ORACLIZE_PAYER, __VA_ARGS__, _self)
+#define oraclize_emplaceQueryId_local(...) __oraclize_emplaceQueryId_local(__VA_ARGS__, _self)
+#define oraclize_getQueryId_local(...) __oraclize_getQueryId_local(__VA_ARGS__, _self)
+
+
+/**************************************************
+ *                PUBLIC FUNCTIONS                *
+ *                 Implementation                 *
+ **************************************************/
+uint64_t oraclize_cbAddress()
+{
+    // go to the connector table which identify the sender 
+    ds_cbaddr cb_addrs("oraclizeconn"_n, "oraclizeconn"_n.value);
+    // point to the first element of the table: 1. if table is empty 
+    auto itr = cb_addrs.begin();
+
+    uint64_t cbaddr = (itr != cb_addrs.end()) ? itr->sender.value : 0;
+    return cbaddr;
+}
+
+std::string vector_to_string(const std::vector<uint8_t> v)
 {
     std::string v_str(v.begin(), v.end());
     return v_str;
 }
 
-std::string checksum256_to_string(checksum256 c)
+std::string capi_checksum256_to_string(const capi_checksum256 c)
```

Comment:
avoid using verbose "capi_" prefix

Reply:
Prefix `capi_` removed for other helpers as well.